base/base/IPv4andIPv6.h

```diff
@@ -62,7 +61,8 @@ namespace std
     {
         size_t operator()(const DB::IPv6 & x) const
         {
-            return std::hash<std::string_view>{}(std::string_view(reinterpret_cast<const char*>(&x.toUnderType()), IPV6_BINARY_LENGTH));
+            return std::hash<std::string_view>{}(
+                std::string_view(reinterpret_cast<const char *>(&x.toUnderType()), sizeof(DB::IPv6::UnderlyingType)));
```

Comment:
Not including `formatIPv6` innecessarily it's a big win, since this file is included pretty much anywhere